src/dataflow/src/source/mod.rs

```diff
@@ -171,6 +171,8 @@ pub struct DecodeResult {
     pub value: Option<Result<Row, DecodeError>>,
     /// The index of the decoded value in the stream
     pub position: Option<i64>,
+    /// The time the record was created in the upstream systsem, as milliseconds since the epoch
+    pub upstream_time_millis: Option<i64>,
```

Comment:
We don't, no. This struct needs a bit more refactoring because sometimes upstream_time_millis and offset also exist in the `metadata` field encoded in the Row